### PR TITLE
micro mix

### PR DESCRIPTION
**Pull request by DavidBJaffe**
- pull in speedups from pat's changes to rust-debrujin (maybe overall 5% faster)
- new system to force uniformity of crate versions
- prevent outputting of parseable field _barcodes if dataset name is null

### CODE REVIEW

File: enclone/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclone"
-version = "0.4.3"
+version = "0.4.2"
```
Comment (wyattmcdonnell) [inline enclone/Cargo.toml:3]:
```suggestion
version = "0.4.3"
```

File: enclone_core/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclone_core"
-version = "0.4.3"
+version = "0.4.2"
```
Comment (wyattmcdonnell) [inline enclone_core/Cargo.toml:3]:
```suggestion
version = "0.4.3"
```

File: enclone_help/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclone_help"
-version = "0.4.3"
+version = "0.4.2"
```
Comment (wyattmcdonnell) [inline enclone_help/Cargo.toml:3]:
```suggestion
version = "0.4.3"
```

File: enclone_main/Cargo.toml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclone_main"
-version = "0.4.3"
+version = "0.4.2"
```
Comment (wyattmcdonnell) [inline enclone_main/Cargo.toml:3]:
```suggestion
version = "0.4.3"
```

File: enclone_print/Cargo.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclone_print"
-version = "0.4.3"
+version = "0.4.2"
```
Comment (wyattmcdonnell) [inline enclone_print/Cargo.toml:3]:
```suggestion
version = "0.4.3"
```

File: enclone_proto/Cargo.toml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclone_proto"
-version = "0.4.3"
+version = "0.4.2"
```
Comment (wyattmcdonnell) [inline enclone_proto/Cargo.toml:3]:
```suggestion
version = "0.4.3"
```

File: enclone_tail/Cargo.toml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclone_tail"
-version = "0.4.3"
+version = "0.4.2"
```
Comment (wyattmcdonnell) [inline enclone_tail/Cargo.toml:3]:
```suggestion
version = "0.4.3"
```

File: enclone_versions/Cargo.toml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "enclone_versions"
-version = "0.4.3"
+version = "0.4.2"
```
Comment (wyattmcdonnell) [inline enclone_versions/Cargo.toml:3]:
```suggestion
version = "0.4.3"
```

**Comment by DavidBJaffe** [comment]
> * pull in speedups from pat's changes to rust-debrujin (maybe overall 5% faster)
> * new system to force uniformity of crate versions
> * prevent outputting of parseable field _barcodes if dataset name is null

**Pull request merged**